content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -0,0 +1,1387 @@
+
+#### Inputs
+- **input** can be connected to a node providing a Float, Int, Vector2, Vector3, Vector4, Matrix, or Geometry type.
+
+#### Outputs
+- **output** is a passthrough of the value connected to input of the same type.
+
+![Geometry info node](/img/tools/nge/geometryInfoNode.jpg)
+
+### GeometryInfo
+This node is also a passthrough node for the Geometry type which adds no time to a graph. It does perform one important task which is supplying access to information about the associated geometry at this specific point of the graph. There may be need to access information like vertex count for debugging purposes which can be fed directly to a debug node. Alternatively, the collectionID of the geometry might be needed to help drive logic in the graph through nodes like equal. This node is flexible enough to fit many different needs which is why it has been added to this group.
+
+#### Inputs
+- **input** can be connected to a node providing a Geometry type.
+
+#### Outputs
+- **output** is a passthrough of the Geometry type connected to input.
+- **id** is an Int representing the ID of the current Geometry.
+- **collectionID** is an Int representing the ID assigned to this Geometry by a collection to which it is assigned.
+- **verticesCount** is an Int representing the total number of vertices in this Geometry.
+- **facesCount** is an Int representing the total number of faces in this Geometry.
+
+![Teleport in node](/img/tools/nge/teleportInNode.jpg)
+
+### TeleportIn
+This node is an organizational node to help reduce the complexity of overlapping wires within a graph. It is a passthrough node that supports Float, Int, Vector2, Vector3, Vector4, Matrix, and Geometry types. This node works in conjunction with the TeleportOut node to make a connection, often far across a graph, without the need for a wire. This node can be connected to any node to create a reference to the value passed by the connected node. When connecting the TeleportIn node to another node, it is important to rename this node to a unique name as this name will be referenced by the TeleportOut node. On the TeleportOut node, selecting the name used by this node as the entry point will create a reference to the value connected to the TeleportIn node. In this way, a single value can be used in several places of a graph without the need to stretch wires across the graph.

```

Comment:
I think you can tell it supports all types of nodes, to be future proof (I guess it supports the soon to be added `Texture` type too?).

Reply:
I wasn't sure if it would support all types, but I guess it likely will so we can say all on this one.

Reply:
fixed in https://github.com/BabylonJS/Documentation/pull/963/commits/5395b92d4e3c0f28c82faf008026b09b521801ce